crates/wast/src/binary.rs

```diff
@@ -1178,6 +1450,51 @@ impl Encode for Names<'_> {
     }
 }
 
+impl ComponentNames<'_> {
+    fn is_empty(&self) -> bool {
+        self.component.is_none()
+            && self.funcs.is_empty()
+            && self.locals.is_empty()
+            && self.labels.is_empty()
+            && self.types.is_empty()
+        // NB: specifically don't check modules/components/instances since they're
+        // not encoded for now.
+    }
+}
+
+impl Encode for ComponentNames<'_> {
```

Comment:
AFAIK there's no spec or thinking yet for how the name section will work out in the binary encoding, so it might be good to ensure that something is written up before implementing something here